lib/trie/database.go

```diff
@@ -305,11 +307,19 @@ func (t *Trie) writeDirtyNode(db Putter, n *Node) (err error) {
 			n.MerkleValue, err)
 	}
 
-	err = db.Put(merkleValue, encoding)
+	if len(merkleValue) < 32 {
+		// Inlined node, there is no need to write it to database.
```

Comment:
could you elaborate a bit more on this statement? I didn't understand it properly 😅. We don't need to write it on DB because it is already part of another encoding that is or will be stored in DB (is that right?)

Reply:
I think this part of the polkadot spec describes it https://spec.polkadot.network/#defn-hashed-subvalue

Reply:
Changed to:

```go
		// Merkle value is the node encoding which is less than 32 bytes.
		// That means this node encoding is inlined in its parent node encoding,
		// and so it is not needed to write it in the database.
```